backend/src/packages/auth/auth.app-plugin.ts

```diff
@@ -0,0 +1,67 @@
+import { type FastifyAuthFunction } from '@fastify/auth';
+import { type FastifyReply, type FastifyRequest } from 'fastify';
+import fp from 'fastify-plugin';
+
+import { HttpMessage } from '~/libs/packages/http/http.js';
+import { type UserEntityObjectWithGroupT } from '~/packages/users/users.js';
+
+import { AuthStrategy } from './auth.js';
+import { assertJwtPayload } from './libs/helpers/assert-jwt-payload.helper.js';
+import { createUnauthorizedError } from './libs/helpers/create-unauthorized-error.helper.js';
+import { type AuthPluginOptions } from './libs/types/auth-plugin-options.type.js';
+
+declare module 'fastify' {
+  interface FastifyInstance {
+    [AuthStrategy.VERIFY_JWT]: FastifyAuthFunction;
+  }
+
+  interface FastifyRequest {
+    user: UserEntityObjectWithGroupT;
+  }
+}
```

Comment:
Create `backend/src/index.d.ts` and move this declaration there. Also update `backend/src/tsconfig.json` - add `"files": ["src/index.d.ts"]`